business/openshift_oauth.go

```diff
@@ -95,14 +87,24 @@ func httpClientWithPool(restConfig rest.Config, systemPool *x509.CertPool) (*htt
 		tlsConfig = &tls.Config{}
 	}
 
-	if restConfig.TLSClientConfig.CAData != nil {
-		log.Trace("Appending CA data from tls client config to pool")
-		if !pool.AppendCertsFromPEM(restConfig.TLSClientConfig.CAData) {
-			return nil, fmt.Errorf("unable to append CA from restConfig to system pool: %s", restConfig.TLSClientConfig.CAData)
-		}
+	// If this setting is set in the Kiali config then override whatever is set on the kubeconfig.
+	if conf.Auth.OpenShift.InsecureSkipVerifyTLS {
+		tlsConfig.InsecureSkipVerify = true
 	}
 
-	tlsConfig.RootCAs = pool
+	if !tlsConfig.InsecureSkipVerify {
+		// Append system certs
+		pool := systemPool.Clone()
+		if restConfig.TLSClientConfig.CAData != nil {
+			log.Trace("Appending CA data from tls client config to pool")
+			if !pool.AppendCertsFromPEM(restConfig.TLSClientConfig.CAData) {
+				return nil, fmt.Errorf("unable to append CA from restConfig to system pool: %s", restConfig.TLSClientConfig.CAData)
+			}
+		}
+		tlsConfig.RootCAs = pool
+	} else {
+		log.Trace("Insecure connection to oAuth server.")
```

Comment:
I can update to `Debug` but you'll see this message every time a user logs into Kiali so it could be very noisy and if you've set the the `Insecure` setting then it shouldn't be a surprise to you that the connection to the `oAuth` server is insecure.

Reply:
I'm more concerned with someone hacking the config or misconfiguring this, allowing the cert validation to be skipped when they aren't expecting it.